src/kessel/grpc/auth.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+
+import google.auth.credentials
+import google.auth.transport.requests
+import requests
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+
+
+class ClientCredentials(google.auth.credentials.Credentials):
```

Comment:
This is a bit generic – there can be lots of credential types. How about `OAuth2ClientCredentials`?